ci: configure `lll` linter to its default value ie 120 chars

humblec opened the pull request:
`lll` linter configuration was made for 180 chars till now due to
long source code signatures/comments we had in the project. This
address it and enabling linter with 120 chars check.
    
 Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
ci: configure `lll` linter to its default value ie 120 chars

`lll` linter configuration was made for 180 chars till now due to
long source code signatures/comments we had in the project. This
address it and enabling linter with 120 chars check.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

nixpanic commented:
please run `make containerized-test` locally and make sure it passes the change to the linter configuration

humblec commented:
@nixpanic up for review :+1:

Madhu-1 commented on scripts/golangci.yml:
```diff
@@ -0,0 +1,170 @@
+---
```
This need to be modified at https://github.com/ceph/ceph-csi/blob/devel/scripts/golangci.yml.in#L113

humblec replied:
@Madhu-1 didnt I modify it already https://github.com/ceph/ceph-csi/pull/2208/commits/a861f252acb44c9b77afd54a5a7efad41583d2eb#diff-2a662b48bcaf012b73a39baf1703a975ab034cb939366529e9fb36c64943fa8fR112? :)

humblec replied:
@Madhu-1 rebased this PR, can you please take a look at this?  this will have conflict coming in each other PR getting in. So the ping.

nixpanic replied:
`scripts/golangci.yml` should not be included, it gets re-generated when `scripts/golangci.yml.in` has been modified.

nixpanic commented on e2e/cephfs.go:
```diff
@@ -298,7 +306,12 @@ var _ = Describe("cephfs", func() {
+					f.ClientSet,
+					f,
+					false,
+					map[string]string{"volumeNamePrefix": volumeNamePrefix},
```
The trailing comma suggests that additional options are possible, but I do not think that `createCephfsStorageClass()` supports that. Is the comma really required?

humblec replied:
sure, removed the `traliing comma` here,

nixpanic reviewed (commented):
Maybe you can document something about coding guidelines? Placing all arguments to many functions make some of the files increase their linecount really a lot. For many, passing a few arguments on the same line would be just as clear without the additional bloat that spreads out the (already long) functions even further.

Madhu-1 commented:
Updating the development guide for future is helpful

humblec committed:
doc: add documentaion about wrapping long lines to coding guide

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
> Maybe you can document something about coding guidelines? Placing all arguments to many functions make some of the files increase their linecount really a lot. For many, passing a few arguments on the same line would be just as clear without the additional bloat that spreads out the (already long) functions even further.

sure, while the line length goes beyond, it has been split, while splitting: placing one by one instead of some in one line and others in next is not better I believe, as the latter is not consistent either. 

@Madhu-1 @nixpanic  i have updated the development guide too. ptal :+1:

nixpanic commented on docs/coding.md:
```diff
@@ -69,6 +69,34 @@ import (
 
+### Wrap long lines
+
+* At present, we restrict the number of chars in a line to `120` which is the
```
no need for the bullet point or indention here, it is the only item in this section of the doc

nixpanic commented on docs/coding.md:
````diff
@@ -69,6 +69,34 @@ import (
+  Example:
+
+```
+  _, err := framework.RunKubectl(cephCSINamespace, "delete", "cm", "ceph-csi-encryption-kms-config", "--namespace", cephCSINamespace, "--ignore-not-found=true")
````
this is not indented either, nor the paragraphs below.

nixpanic commented on cmd/cephcsi.go:
```diff
@@ -66,26 +66,66 @@ func init() {
+		&conf.HistogramOption,
+		"histogramoption",
+		"0.5,2,6",
+		"[DEPRECATED] Histogram option for grpc metrics, should be comma separated value,"+
```
missing space after the comma inside the string at `,"`

nixpanic commented on internal/cephfs/clone.go:
```diff
@@ -134,7 +134,11 @@ func createCloneFromSubvolume(ctx context.Context, volID, cloneID volumeID, volO
+func cleanupCloneFromSubvolumeSnapshot(
+	ctx context.Context,
+	volID, cloneID volumeID,
+	parentVolOpt *volumeOptions,
```
is the comma at the end really required? it looks so incomplete when used with function calls

nixpanic commented:
In general for PRs of this type, we recommend to send multiple small PRs. A PR for a package like `internal/` is not practical, it will be easier to split it up in PRs per sub-package, or even per (group of) files. That way, it will be easier for others to review, reduce the number of conflicts you need to correct in this PR, and make it easier for other developers to rebase their PRs after parts have been merged.

humblec commented:
> In general for PRs of this type, we recommend to send multiple small PRs. A PR for a package like `internal/` is not practical, it will be easier to split it up in PRs per sub-package, or even per (group of) files. That way, it will be easier for others to review, reduce the number of conflicts you need to correct in this PR, and make it easier for other developers to rebase their PRs after parts have been merged.

Sure, I can split large commits to groups or subpackage level.. Let me revisit this

humblec committed:
internal: reformat long lines in internal/cephfs package to 120 chars

We have many declarations and invocations..etc
with long lines which are very difficult to follow while doing
code reading. This address the issues in 'internal/cephfs' package to
restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
internal: reformat long lines in internal/controller package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'internal/controller' package to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
internal: reformat long lines in internal/csi-common package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'internal/csi-common' package to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
internal: reformat long lines in internal/journal package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'internal/journal' package to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
internal: reformat long lines in internal/rbd package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'internal/rbd/*server.go' and 'internal/rbd/driver.go' files to restrict
the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
internal: reformat long lines in internal/rbd package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in below files, and restrict the line length to 120 chars.

-internal/rbd/rbd_attach.go
-internal/rbd/rbd_journal.go
-internal/rbd/rbd_util.go
-internal/rbd/replicationcontrollerserver.go
-internal/rbd/snapshot.go

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
internal: reformat long lines in internal/util package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'internal/util' package files to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
cephfs: reformat long lines in 'cmd' package to 120 chars

We have many declarations and invocations..etc
with long lines which are very difficult to follow while doing
code reading. This address the issues in 'cmd' package to restrict
the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/cephfs*.go' files to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/rbd*.go' files to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/rbd_helper.go' files to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/deploy_vault.go' file to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/pod.go' files to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/pvc.go' file to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/resize.go' files to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/utils.go' files to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/staticpvc.go' and 'e2e/snapshot.go' files to restrict the line length
to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: reformat long lines in this package to 120 chars

We have many declarations and invocations..etc with long lines which are
very difficult to follow while doing code reading. This address the issues
in 'e2e/upgrade_*.go' files to restrict the line length to 120 chars.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
@nixpanic all comments are addressed and commits are also organized. PTAL :+1:

humblec commented:
@Madhu-1 ptal

Madhu-1 commented on cmd/cephcsi.go:
```diff
@@ -66,26 +66,58 @@ func init() {
 	flag.BoolVar(&conf.IsNodeServer, "nodeserver", false, "start cephcsi node server")
-	flag.StringVar(&conf.DomainLabels, "domainlabels", "", "list of kubernetes node labels, that determines the topology"+
-		" domain the node belongs to, separated by ','")
+	flag.StringVar(
```
Commit description is not matching the changes , can you please revisit this one and cross check other commits also

humblec commented:
@Madhu-1 corrected, rest of the commit looks good .. Thanks .

Yuggupta27 commented on e2e/rbd.go:
```diff
@@ -331,7 +338,13 @@ var _ = Describe("RBD", func() {
+					defaultSCName,
+					nil,
+					map[string]string{"csi.storage.k8s.io/fstype": "ext4"},
+					deletePolicy)
```
Since we are planning to split the parameters into different lines, It becomes difficult to read as it is harder to find the end of the function call. 
Suggested change(here and everywhere else):
In place of closing the function call with 
```
deletePolicy)
```
Closing it out in a separate line will make it more readable.
```
deletePolicy
)
```


E.g. 
```
err = createRBDStorageClass(
					f.ClientSet,
					f,
					defaultSCName,
					nil,
					map[string]string{"csi.storage.k8s.io/fstype": "ext4"},
					deletePolicy
                           )
```
Looks more clear to me than:
```
err = createRBDStorageClass(
					f.ClientSet,
					f,
					defaultSCName,
					nil,
					map[string]string{"csi.storage.k8s.io/fstype": "ext4"},
					deletePolicy)
```

Thoughts?

nixpanic replied:
Yes, when placing all arguments for a function on separate lines, sufficient indenting (automated by `go fmt`) and the closing `)` on a separate line too makes sense to me.

nixpanic commented on e2e/pod.go:
```diff
@@ -128,7 +137,11 @@ func findPodAndContainerName(f *framework.Framework, ns, cn string, opt *metav1.
+func getCommandInPodOpts(
+	f *framework.Framework,
+	c, ns, cn string,
+	opt *metav1.ListOptions,
```
this is still using the last-argument-has-trailing-comma style

nixpanic reviewed (dismissed):
There is currently a mix of

```go
func bla(
    arg int,
    s string,
) error {
```

and

```go
func bla(
    arg int,
    s string) error {
```

I really do not like the trailing comma for function arguments (it is fine when filling a struct or map). So, ideally I prefer the format

```go
func bla(
    arg int,
    s string
) error {
```

but as that was not used anywhere, maybe Golang (or `go fmt`) does not allow it? If that is the case, I would opt for

```go
func bla(
    arg int,
    s string) error {
```

This is a **-1** only because there is isn't a single style used consequently. When changing the style, it should be the same everywhere.

Madhu-1 commented:
> There is currently a mix of
> 
> ```go
> func bla(
>     arg int,
>     s string,
> ) error {
> ```
> 
> and
> 
> ```go
> func bla(
>     arg int,
>     s string) error {
> ```
> 
> I really do not like the trailing comma for function arguments (it is fine when filling a struct or map). So, ideally I prefer the format
> 
> ```go
> func bla(
>     arg int,
>     s string
> ) error {
> ```
> 
> but as that was not used anywhere, maybe Golang (or `go fmt`) does not allow it? If that is the case, I would opt for
> 
> ```go
> func bla(
>     arg int,
>     s string) error {
> ```
> 
> This is a **-1** only because there is isn't a single style used consequently. When changing the style, it should be the same everywhere.

The build will fail for the below format
```
func bla(
    arg int,
    s string
) error {
```
it should be
```
func bla(
    arg int,
    s string,
) error {
```

humblec commented:
> ```
> func bla(
>     arg int,
>     s string
> ) error {
> ```
> 
> it should be
> 
> ```
> func bla(
>     arg int,
>     s string,
> ) error {
> ```

@nixpanic @Yuggupta27  @Madhu-1 Above was the style in `very first version of the PR`, then changed everything to avoid trailing comma in last arg of functions as per previous discussions and `)` parenthesis brought back to last arg line along with `returns`. Ideally, above is ( which was the first version) is what I have seen in other projects. I dont see a reason for not following that.

Are we in agreement for above version ? 

or we want something like below?

```
func blah(
arg int,
s string) error {
...
```

please ack any of this , its really difficult to keep changing different formats in each version of the PR!

nixpanic commented:
Functions are not expected to change the number of arguments often, so I would prefer

```go
func blah(
    arg int,
    s string) error {
...
```

for the definition, as well as the places where they get called:

```go
                                        err := blah(
                                            arg,
                                            s)
```

But obviously only when there is a need, and the line length crosses the limit.

humblec commented:
@nixpanic @Madhu-1 @Yuggupta27 ptal .

nixpanic commented on e2e/pvc.go:
```diff
@@ -141,7 +154,10 @@ func deletePVCAndPV(c kubernetes.Interface, pvc *v1.PersistentVolumeClaim, pv *v
-func getPVCAndPV(c kubernetes.Interface, pvcName, pvcNamespace string) (*v1.PersistentVolumeClaim, *v1.PersistentVolume, error) {
+func getPVCAndPV(
+	c kubernetes.Interface,
+	pvcName, pvcNamespace string,
```
:bell: missed one

humblec replied:
@nixpanic corrected this too :),, thanks . ptal

nixpanic reviewed (dismissed):
argh, **so close**!

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.20

nixpanic commented:
> /test ci/centos/mini-e2e/k8s-1.20

poking Jenkins after service outage

humblec commented:
/retest all
